src/lib/authui/main_dialog.rb

```diff
@@ -0,0 +1,120 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE Linux GmbH.
+#
+# ------------------------------------------------------------------------------
+# Authors:      Howard Guo <hguo@suse.com>
+#
+
+require 'yast'
+require 'ui/dialog'
+require 'auth/authconf'
+require 'authui/sssd/main_dialog'
+require 'authui/ldapkrb/main_dialog'
+Yast.import 'UI'
+Yast.import 'Icon'
+Yast.import 'Label'
+
+module Auth
+    # Main dialog displays an overview of authentication mechanisms enabled on the system.
+    class MainDialog < UI::Dialog
+        include Yast
+        include Auth
+        include UIShortcuts
+        include I18n
+        include Logger
+
+        # Entry point can be :sssd, :ldapkrb, or :auto
+        # In auto mode, there will be two change settings buttons.
+        def initialize(entry_point)
+            super()
+            textdomain 'auth-client'
+            @entry_point = entry_point
+        end
+
+        def dialog_options
+            Opt(:decorated, :defaultsize)
+        end
+
+        def create_dialog
+            return false unless super
+            render_info_table
+            return true
+        end
+
+        def dialog_content
+            conf_buttons = [PushButton(Id(:change_settings), _('Change Settings')), PushButton(Id(:cancel), Label.CancelButton)]
+            if @entry_point == :auto
+                # Allow entering both SSSD and ldapkrb settings
+                conf_buttons = [
+                    PushButton(Id(:change_sssd_settings), _('User Logon Configuration')),
+                    PushButton(Id(:change_ldapkrb_settings), _('LDAP/Kerberos Configuration')),
+                    PushButton(Id(:cancel), Label.FinishButton)
+                ]
+            end
+            Left(HBox(
+                HWeight(10, Empty()),
+                HWeight(80, VBox(
+                    Left(Frame(_('System authentication and domain configuration'), VBox(
```

Comment:
Where should the change button be placed, if exit button is placed at bottom right?